File: CS/SalaryReports - Step 07/MainForm.cs
using System.Windows.Forms;
using SalaryReports.UserControls;

namespace SalaryReports
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void windowsUIView1_QueryControl(object sender, DevExpress.XtraBars.Docking2010.Views.QueryControlEventArgs e)
        {
            switch ((int)e.Document.Tag)
            {
                case 0:
                    e.Control = new DocumentViewerUserControl() { Report = new ManagementReport() };
                    break;
                case 1:
                    e.Control = new DocumentViewerUserControl() { Report = new EmployeeReport() };
                    break;
                case 2:
                    EmployeeReport report = new EmployeeReport();
                    report.Parameters["paramEmpName"].Value = string.Empty;
                    report.Parameters["paramEmpName"].Visible = false;
                    e.Control = new DocumentViewerUserControl() { Report = report };
                    break;
            }
        }
    }
}

File: VB/SalaryReports - Step 07/MainForm.vb
Imports System.Windows.Forms
Imports SalaryReports.UserControls

Namespace SalaryReports

    Public Partial Class MainForm
        Inherits System.Windows.Forms.Form

        Public Sub New()
            Me.InitializeComponent()
        End Sub

        Private Sub windowsUIView1_QueryControl(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Docking2010.Views.QueryControlEventArgs)
            Select Case CInt(e.Document.Tag)
                Case 0
                    e.Control = New SalaryReports.UserControls.DocumentViewerUserControl() With {.Report = New SalaryReports.ManagementReport()}
                Case 1
                    e.Control = New SalaryReports.UserControls.DocumentViewerUserControl() With {.Report = New SalaryReports.EmployeeReport()}
                Case 2
                    Dim report As SalaryReports.EmployeeReport = New SalaryReports.EmployeeReport()
                    report.Parameters(CStr(("paramEmpName"))).Value = String.Empty
                    report.Parameters(CStr(("paramEmpName"))).Visible = False
                    e.Control = New SalaryReports.UserControls.DocumentViewerUserControl() With {.Report = report}
            End Select
        End Sub
    End Class
End Namespace
